editor/src/clj/editor/render_target.clj

```diff
@@ -0,0 +1,129 @@
+            [editor.gl.texture :as texture]
+            [editor.protobuf :as protobuf]
+            [editor.protobuf-forms :as protobuf-forms]
+            [editor.resource-node :as resource-node]
+            [editor.workspace :as workspace])
+  (:import [com.dynamo.graphics.proto Graphics$TextureImage$Type Graphics$TextureImage$TextureFormat]
+           [com.dynamo.render.proto RenderTarget$RenderTargetDesc]))
+
+(def ^:const atlas-icon "icons/32/Icons_13-Atlas.png")
+
+(defn- set-form-op [{:keys [node-id]} [property] value]
+  (g/set-property! node-id property value))
+
+(defn- clear-form-op [{:keys [node-id]} [property]]
+  (g/clear-property! node-id property))
+
+(def form-data
+  {:navigation false
+   :sections
+   [{:title "Render Target"
+     :fields [{:path [:color-attachments]
+               :label "Color Attachments"
+               :type :table
+               :columns [{:path [:width]
+                          :label "width"
+                          :type :number}
+                         {:path [:height]
+                          :label "height"
+                          :type :number}
+                         {:path [:format]
+                          :label "format"
+                          :type :choicebox
+                          :options (protobuf-forms/make-options (protobuf/enum-values Graphics$TextureImage$TextureFormat))
+                          :default (ffirst (protobuf/enum-values Graphics$TextureImage$TextureFormat))}]}
+              {:path [:depth-stencil-attachment-width]
+               :label "Depth/Stencil Width"
+               :type :number}
+              {:path [:depth-stencil-attachment-height]
+               :label "Depth/Stencil Height"
+               :type :number}
+              {:path [:depth-stencil-attachment-texture-storage]
+               :label "Depth Buffer Texture Storage"
+               :type :boolean}]}]})
+(g/defnk produce-form-data [_node-id color-attachments depth-stencil-attachment-width depth-stencil-attachment-height depth-stencil-attachment-texture-storage :as args]
+  (let [values (select-keys args (mapcat :path (get-in form-data [:sections 0 :fields])))
+        form-values (into {} (map (fn [[k v]] [[k] v]) values))]
+    (-> form-data
+        (assoc :values form-values)
+        (assoc :form-ops {:user-data {:node-id _node-id}
+                          :set set-form-op
+                          :clear clear-form-op}))))
+
+(g/defnk produce-pb-msg
+  [color-attachments depth-stencil-attachment-width depth-stencil-attachment-height depth-stencil-attachment-format depth-stencil-attachment-texture-storage]
+  {:color-attachments color-attachments
+   :depth-stencil-attachment {:width depth-stencil-attachment-width
+                              :height depth-stencil-attachment-height
+                              :format depth-stencil-attachment-format
+                              :texture-storage depth-stencil-attachment-texture-storage}})
```

Comment:
This doesn't seem to match the updated structure of `RenderTarget$RenderTargetDesc`. How can this possibly work? 🤔

Reply:
Oh, never mind. I just got confused... 😵‍💫
This seems fine after all.